src/main/java/org/jenkins/plugins/lockableresources/LockableResource.java

```diff
@@ -125,9 +128,21 @@ protected Object readResolve() {
     if (queuedContexts == null) { // this field was added after the initial version if this class
       queuedContexts = new ArrayList<>();
     }
+    this.repairLabels();
     return this;
   }
 
+  @Restricted(NoExternalUse.class)
+  public void repairLabels() {
+    if (this.labels == null) {
+      return;
+    }
+
+    LOGGER.fine("Repair labels for resource " + this);
+    this.setLabels(this.labels);
+    this.labels = null;
+  }
+
```

Comment:
This code can be inlined

Reply:
how?

Reply:
I mean no extra method, just some more statements in `readResolve`, then you have everything in one place. But this is a matter of taste.